### PR TITLE
chore(ooo): ooo c code optimisation for different CPU architectures

### PR DESCRIPTION
**Pull request by ideoma**

### CODE REVIEW

File: benchmarks/src/main/java/org/questdb/OooCppBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+public class OooCppBenchmark {
+    private static final long BUFFER_MAX_SIZE = 256 * 1024 * 1024L;
+    private static long buffer;
+    private static final long MB = 1024 * 1024L;
+
+    public static void main(String[] args) throws RunnerException, SqlException, IOException {
+
+        Os.init();
+        System.out.println("MemorySuffle");
```
Comment (bluestreak01) [inline benchmarks/src/main/java/org/questdb/OooCppBenchmark.java:46]:
```suggestion
        System.out.println("MemoryShuffle");
```

File: benchmarks/src/main/java/org/questdb/OooCppBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        testSetMemoryShuffleToCsv();
+
+//        System.out.println("SetMemoryVanilla");
```
Comment (bluestreak01) [inline benchmarks/src/main/java/org/questdb/OooCppBenchmark.java:49]: remove?

File: core/CMakeLists.txt [hunk 3]
```diff
@@ -213,6 +218,13 @@ else ()
             COMPILE_FLAGS "-m64 -O3 "
     )
 
+    ## TODO: SSE2 and SSE4 are redundant on 64 bit CPUs
```
Comment (bluestreak01) [inline core/CMakeLists.txt:221]: shall we then remove SSE2?

File: core/src/main/c/share/ooo.cpp [hunk 4]
```diff
@@ -238,6 +226,7 @@ inline void swap(index_t *a, index_t *b) {
  *  of pivot
  *
  **/
+__SIMD_MULTIVERSION__
```
Comment (bluestreak01) [inline core/src/main/c/share/ooo.cpp:229]: we are not using GCC multiversion? should we perhaps tidy this up?

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: benchmarks/src/main/java/org/questdb/OooCppBenchmark.java (hunk 1 above)
Comment (ideoma) [reply benchmarks/src/main/java/org/questdb/OooCppBenchmark.java:46]: ok

File: benchmarks/src/main/java/org/questdb/OooCppBenchmark.java (hunk 2 above)
Comment (ideoma) [reply benchmarks/src/main/java/org/questdb/OooCppBenchmark.java:49]: separated to another class, can be useful for further optimisations

File: core/CMakeLists.txt (hunk 3 above)
Comment (ideoma) [reply core/CMakeLists.txt:221]: I checked again and found that SSE4 is not defined with vanilla compilation so I'm removing the TODO

File: core/src/main/c/share/ooo.cpp (hunk 4 above)
Comment (ideoma) [reply core/src/main/c/share/ooo.cpp]: ok

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**